impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.h"
+
+#include "flutter/fml/trace_event.h"
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/vulkan/barrier_vk.h"
+#include "impeller/renderer/backend/vulkan/command_buffer_vk.h"
+#include "impeller/renderer/backend/vulkan/command_encoder_vk.h"
+#include "impeller/renderer/backend/vulkan/swapchain/ahb/ahb_formats.h"
+#include "impeller/renderer/backend/vulkan/swapchain/surface_vk.h"
+#include "impeller/toolkit/android/surface_transaction.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// The maximum number of presents pending in the compositor after which the
+/// acquire calls will block.
+///
+static constexpr const size_t kMaxPendingPresents = 2u;
```

Comment:
With a max pending presents of 2 I think the max image count should be three, since that is two used by the compositor and 1 being worked on by flutter.

Reply:
FYI @chinmaygarde . Not blocking since we could adjust to 3 later, but IMO 3 makes then most sense